### PR TITLE
Demonstrate lockfile support

### PR DESCRIPTION
**Pull request by Eric-Arellano**
In https://pants.readme.io/docs/python-third-party-dependencies, we now explain why and how to use a lockfile. This demonstrates actually doing this.

In the process, we demonstrate creating a virtual environment, which is needed for things like editor integration.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,141 +2,14 @@
...
+/.pants.workdir.file_lock*
 
-# Byte-compiled / optimized / DLL files
+# Python files
 __pycache__/
-*.py[cod]
-*$py.class
+/.venv/
 
-# C extensions
```
Comment (Eric-Arellano) [inline .gitignore:15]: These were noisy. We don't plan to use things like C extensions or `nosetests.xml`.

File: constraints.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Pants project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# Generated by first creating a virtual environment with `python3.7 -m venv .venv`, then running
+# `.venv/bin/pip install -r requirements.txt` and `.venv/bin/pip freeze`.
```
Comment (Eric-Arellano) [inline constraints.txt:5]: I don't know if we should call more attention to how to create a venv? For example, should we add this to the README or make a simple script like `mk_venv.sh`?

File: .gitignore (hunk 1 above)
Comment (benjyw) [reply .gitignore:15]: We still need `.pyc` for python2 no?

File: constraints.txt (hunk 2 above)
Comment (benjyw) [reply constraints.txt:5]: This is fine for now, I think.

File: .gitignore (hunk 1 above)
Comment (Eric-Arellano) [reply .gitignore:15]: I think it should be covered, but I added it back to be explicit.

Comment (benjyw) [reply .gitignore:15]: Covered by what?

Comment (Eric-Arellano) [reply .gitignore:15]: Covered by `__pycache__`. I often use this in projects and it seems to work. But, I don't know how it behaves with Python 2, so I just added `*.pyc` too.

Comment (benjyw) [reply .gitignore:15]: AFAIK python2 writes .pyc files as siblings of the .py files, not in `__pycache__`?

**Pull request merged**